File: python/java_to_python/09_pgm_avion_interface/TipoCarga.py
import time
from Avion import Avion
from IAvionCarga import IAvionCarga

class TipoCarga(Avion, IAvionCarga):
    def __init__(self):
        super().__init__()
        self.setTipo("Carga")
    
    def startingEngine(self):
        print("⛽Check Fuel ✅")
        for i in range(3):
            print("💥💥Spark💥💥")
            try:
                time.sleep(0.8 - (i * 0.4))
            except Exception as e:
                print("⛔: {}".format(e))
        print("Started the Motor...✈️✈️")
    
    def speedUp(self):
        speedUpString = "speed up in Progress: "
        for i in range(self.getVelocidad()):
            speedUpString += "✈️"
        
        for i in range(10):
            speedUpString += "✈️"
            print(speedUpString)
            try:
                time.sleep(0.5 - (i * 0.05))
            except Exception as e:
                print("⛔: {}".format(e))
        self.addSpeed(10)
    
    def stopingEngine(self):
        print("✅ Check the Turbine 🆗🆗")
        for i in range(2):
            print("⛔ turning off turbine❌❌")
            try:
                time.sleep(0.8 - (i * 0.4))
            except Exception as e:
                print("⛔: {}".format(e))
        print("Stoped...⚡⚡")
    
    def openDoors(self):
        print("✅ Open Door 🆗")
    
    def closeDoors(self):
        print("✅ Close Door 🆗")
    
    def pickUp(self):
        print("✅ PickUp 🆗")
    
    def chargeFuel(self, litros):
        super().chargeFuel(litros)
        print("")
        for i in range(litros):
            try:
                print("⛽", end="")
                time.sleep(0.1)
            except Exception as e:
                print("⛔: {}".format(e))
        print("")
        print("⛽ Full Fuel ✅")

File: ruby/pgms_ruby/09-pgmAvionInterface/TipoCarga.rb
require_relative 'Avion'
require_relative 'IAvionCarga'

class TipoCarga < Avion
  include IAvionCarga
  
  def initialize
    super
    self.setTipo("Carga")
  end

  def startingEngine
    puts "⛽Check Fuel ✅"
    3.times do |i|
      print "💥💥Spark💥💥"
      begin
        sleep(0.8 - (i * 0.4))
      rescue Interrupt
        puts "⛔: KeyboardInterrupt"
      rescue Exception => e
        puts "⛔: #{e}"
      end
    end
    puts "Started the Motor...✈️✈️"
  end

  def speedUp
    speedUpString = "speed up in Progress: "
    super.getVelocidad.times do
      speedUpString += "✈️"
    end

    10.times do |i|
      speedUpString += "✈️"
      puts speedUpString
      begin
        sleep(0.5 - (i * 0.05))
      rescue Interrupt
        puts "⛔: KeyboardInterrupt"
      rescue Exception => e
        puts "⛔: #{e}"
      end
    end
    super.addSpeed(10)
  end

  def stopingEngine
    puts "✅ Check the Turbine 🆗🆗"
    2.times do |i|
      puts "⛔ turning off turbine❌❌"
      begin
        sleep(0.8 - (i * 0.4))
      rescue Interrupt
        puts "⛔: KeyboardInterrupt"
      rescue Exception => e
        puts "⛔: #{e}"
      end
    end
    puts "Stoped...⚡⚡"
  end

  def openDoors
    puts "✅ Open Door 🆗"
  end

  def closeDoors
    puts "✅ Close Door 🆗"
  end

  def pickUp
    puts "✅ PickUp 🆗"
  end

  def chargeFuel(litros)
    super.chargeFuel(litros)
    puts ""
    litros.times do
      begin
        print "⛽"
        sleep(0.1)
      rescue Interrupt
        puts "⛔: KeyboardInterrupt"
      rescue Exception => e
        puts "⛔: #{e}"
      end
    end
    puts ""
    puts "⛽ Full Fuel ✅"
  end
end